packages/components/src/text-highlight/test/index.tsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import TextHighlight from '..';
+
+const getMarks = ( container: Element ) =>
+	Array.from( container.querySelectorAll( 'mark' ) );
+
+const defaultText =
+	'We call the new editor Gutenberg. The entire editing experience has been rebuilt for media rich pages and posts.';
+
+describe( 'TextHighlight', () => {
+	describe( 'Basic rendering', () => {
+		it.each( [ [ 'Gutenberg' ], [ 'media' ] ] )(
+			'should highlight the singular occurance of the text "%s" in the text if it exists',
+			( highlight ) => {
+				const { container } = render(
+					<TextHighlight
+						text={ defaultText }
+						highlight={ highlight }
+					/>
+				);
+
+				const highlightedEls = getMarks( container );
+
+				highlightedEls.forEach( ( el ) => {
+					expect( el.textContent ).toEqual(
+						expect.stringContaining( highlight )
+					);
```

Comment:
Just curious: Have you considered using [`.toHaveTextContent()`](https://github.com/testing-library/jest-dom#tohavetextcontent) for these? I find it a bit more ergonomic and less error-prone, but maybe it's just personal preference.

```suggestion
					expect( el ).toHaveTextContent(
						new RegExp( `^${ highlight }$` )
					);
```